core/vocabulary/vocabulary.py

```diff
@@ -208,40 +185,83 @@ def _add_selection_to_csv(
         is_acronym = re.fullmatch(r"[A-Z]+", written_form)
         spoken_form = " ".join(written_form) if is_acronym else written_form
     entries = _create_vocabulary_entries(spoken_form, written_form, type)
-    new_entries = {}
     added_some_phrases = False
-    for spoken_form, written_form in entries.items():
-        if skip_identical_replacement and spoken_form == written_form:
-            actions.app.notify(f'Skipping identical replacement: "{spoken_form}"')
-        elif spoken_form in csv_contents:
-            actions.app.notify(f'Spoken form "{spoken_form}" is already in {csv}')
-        else:
-            new_entries[spoken_form] = written_form
-            added_some_phrases = True
-    append_to_csv(csv, new_entries)
+
+    # until we add support for parsing or otherwise getting the active
+    # vocabulary.talon-list, skip the logic for checking for duplicates etc
+    if file_contents:
+        # clear the new entries dictionary
+        new_entries = {}
+        for spoken_form, written_form in entries.items():
+            if skip_identical_replacement and spoken_form == written_form:
+                actions.app.notify(f'Skipping identical replacement: "{spoken_form}"')
+            elif spoken_form in file_contents:
+                actions.app.notify(
+                    f'Spoken form "{spoken_form}" is already in {file_name}'
+                )
+            else:
+                new_entries[spoken_form] = written_form
+                added_some_phrases = True
+    else:
+        new_entries = entries
+        added_some_phrases = True
+
+    if file_name.endswith(".csv"):
+        append_to_csv(file_name, new_entries)
+    elif file_name == "vocabulary.talon-list":
+        append_to_vocabulary(new_entries)
+
     if added_some_phrases:
-        actions.app.notify(f"Added to {csv}: {new_entries}")
+        actions.app.notify(f"Added to {file_name}: {new_entries}")
+
+
+def append_to_vocabulary(rows: dict[str, str]):
+    vocabulary_file_path = actions.user.get_vocabulary_file_path()
+    with open(str(vocabulary_file_path)) as file:
+        line = None
+        for line in file:
+            pass
+        needs_newline = line is not None and not line.endswith("\n")
+
+    with open(vocabulary_file_path, "a", encoding="utf-8", newline="") as file:
+        if needs_newline:
+            file.write("\n")
+        for key, value in rows.items():
+            if key == value:
+                file.write(f"{key}")
+            else:
+                value = repr(value)
```

Comment:
We had decided to only use repr or another quoting mechanism if absolutely required (e.g., matching the value with a [A-Za-z] regex to be extremely conservative, or match the actual behavior of the .talon-list parser).

Reply:
Done my best to match the actual behavior, at least for non-pathological cases.